src/jsct.js

```diff
@@ -65,13 +66,10 @@ program.command('parse')
 
 program.command('new')
   .description('create a WebAssembly component adapted from a component core Wasm [wasm-tools component new]')
-  .argument('[module]', 'Wasm core module filepath')
+  .argument('[core-module]', 'Wasm core module filepath')
   .requiredOption('-o, --output <output-file>', 'Wasm component output filepath')
   .option('--name <name>', 'custom output name')
-  .option('--wit <wit>', 'WIT file to use')
```

Comment:
FWIW this argument is still useful to have but for the `wasm-tools` CLI I moved it to a new `wasm-tools component embed` subcommand which is explicitly only about embedding information in a core wasm module which is then later fed to `wasm-tools component new`. Similar to the `--encoding` flag that's where those moved to if you're intersted in having an `embed` command here as well.

Although to some degree if you're interested I think it would be reasonable to expose the `wasm-tools` CLI as a clap-based library. That way you wouldn't have to rewrite the CLI here if you'd like.

Reply:
I'd like to implement the `embed` command and will need to integrate the `dummy` support as well. It's certainly a bunch of wrapper work, but I think it's useful that it can run in any JS environment including a web browser without it being tied to a CLI / WASI world. That said, with the new resolver stuff that will certainly depend on WASI I believe. Hoping to iterate through this further with the preview2 update.

Reply:
Sure, let me see how the WASI integration goes here first. On the one hand it's nice not to have to write to temporary files for in-memory ops (eg we do currently support `import { componentNew } from 'js-component-tools'` being fully in-memory), but with the latest resolver systems we will need WASI as a dependency now anyway.

Reply:
To clarify, though, all of that is supported by the libraries of `wit-parser`, `wit-component`, and the `wasm-tools` crates. I'm mostly just putting this out there of there's possible more code to reuse if you'd like.

Reply:
At the very least we need a component build of the library interfaces though to achieve those use cases in a JS tool without requiring the Rust toolchain.